### PR TITLE
[IPEX] Fix native windows setup

### PR DESCRIPTION
**Pull request by Nuullll**
## Description

This is a W/A for IPEX native windows. The latest `torchvision` will install `torch 2.0.1` which is incompatible with `ipex`, so we double confirm the intel torch wheel (`2.0.0a0`) is installed after installing all packages.

## Notes

Related issues: https://github.com/vladmandic/automatic/issues/1955, https://github.com/intel/intel-extension-for-pytorch/issues/398

## Environment and Testing

Tested with Intel Arc A770 16G (windows driver: 31.0.101.4502) + Win11

194dd7f2ec883c0fc8de93407458a0a2

### CODE REVIEW

**Comment by Nuullll** [comment]
@Disty0

File: installer.py [hunk 1]
```diff
@@ -287,6 +287,30 @@ def check_python():
...
+# Intel hasn't released a corresponding torchvision wheel along with torch and
+# ipex wheels, so we have to install official pytorch torchvision as a W/A.
+# However, the latest torchvision explicitly requires torch version == 2.0.1,
+# which is incompatible with the Intel torch version 2.0.0a0. This will cause
+# intel torch to be uninstalled when pip scans the dependencies of torchvision.
+# This function will check the torch version and force installing Intel torch
+# 2.0.0a0 to avoid the underlying dll version error.
+# TODO remove this W/A when Intel releases torchvision wheel for windows.
+def fix_ipex_win_torch():
+    if not args.use_ipex or not 'win' in sys.platform:
```
Comment (Nuullll) [inline installer.py:299]:
```suggestion
    if not args.use_ipex or 'win' not in sys.platform:
```

**Comment by vladmandic** [comment]
creating a full function `fix_ipex_win_torch` in installer seems unnecessary when it could be two lines placed in a proper place in the code and there is already other platform specific code (e.g. rocm) that does it that way.

**Comment by Nuullll** [comment]
> creating a full function `fix_ipex_win_torch` in installer seems unnecessary when it could be two lines placed in a proper place in the code and there is already other platform specific code (e.g. rocm) that does it that way.

Thanks for reviewing!
Could you please clarify "in a proper place in the code"? If you mean [`ipex_specific`](https://github.com/vladmandic/automatic/tree/ff2d846667f62bd83a49b85e44e137e97c64727b/modules/ipex_specific), then I'm unsure whether it's a good idea to involve installing commands inside `modules.devices`.

The problem is that everytime `pip` installs a package that depends on `torchvision`, then `torch == 2.0.1` will be installed (overriding `torch == 2.0.0a0`). So I attempted to ensure a correct `torch` version in two places:
- after all `pip` commands are run
- right before `import torch` during installation

**Comment by Disty0** [comment]
Since ROCm has it's own installer, should we push this too? @vladmandic

**Comment by vladmandic** [comment]
> Since ROCm has it's own installer, should we push this too? @vladmandic I don't see Intel releasing a proper package stack anytime soon.

you mean include the `fix_ipex` in the installer? sure, why not.

**Pull request merged**